File: src/Domestica.Budget.Application/Transactions/AddOutcomeTransaction/AddOutcomeTransactionCommand.cs
using CommonAbstractions.DB.Messaging;
using Domestica.Budget.Domain.Transactions;

namespace Domestica.Budget.Application.Transactions.AddOutcomeTransaction
{
    public sealed record AddOutcomeTransactionCommand(
        string SourceAccountId,
        string RecipientName,
        decimal TransactionAmount,
        string CategoryValue,
        string SubcategoryValue,
        DateTime TransactionDateTime) : ICommand<Transaction>
    {
    }
}

File: src/client-app/src/models/requests/addOutcomeTransactionCommand.ts

export interface AddOutcomeTransactionCommand {
    sourceAccountId: string
    recipientName: string
    transactionAmount: number
    categoryValue: string
    transactionDateTime: Date
}

export class AddOutcomeTransactionCommand implements AddOutcomeTransactionCommand {
    constructor(values?: AddOutcomeTransactionCommand) {
        this.sourceAccountId = values?.sourceAccountId || ''
        this.recipientName = values?.recipientName || ''
        this.transactionAmount = values?.transactionAmount || 0
        this.categoryValue = values?.categoryValue || ''
        this.transactionDateTime = values?.transactionDateTime || new Date()
    }
}